AdminServer/appscale/admin/__init__.py

```diff
@@ -98,30 +98,32 @@ def wait_for_port_to_open(http_port, operation_id, deadline):
   except KeyError:
     raise OperationTimeout('Operation no longer in cache')
 
+  all_lbs = set(appscale_info.get_load_balancer_ips())
+  passed_lbs = set()
   while True:
-    if time.time() > deadline:
-      message = 'Deploy operation took too long.'
-      operation.set_error(message)
-      raise OperationTimeout(message)
-
-    if utils.port_is_open(options.login_ip, http_port):
-      break
+    for load_balancer in all_lbs:
+      if load_balancer in passed_lbs or time.time() > deadline:
+        continue
 
-    yield gen.sleep(1)
+      if utils.port_is_open(load_balancer, http_port):
+        passed_lbs.add(load_balancer)
 
-  for load_balancer in appscale_info.get_load_balancer_ips():
-    while True:
-      if time.time() > deadline:
-        # The version is reachable from the login IP, but it's not reachable
-        # from every registered load balancer. It makes more sense to mark the
-        # operation as a success than a failure because the lagging load
-        # balancers should eventually reflect the registered instances.
-        break
+    if len(passed_lbs) == len(all_lbs):
```

Comment:
This would prevent the system to operate in case one of the LB failed: I don't think we remove a failed LBs from the list of nodes yet, do we?

Reply:
It means we do not proceed until the deadline is reached, which I believe is the same as the current behaviour.